airflow/models/skipmixin.py

```diff
@@ -121,14 +128,12 @@ def skip(
                 stacklevel=2,
             )
 
-            dag_run = (
-                session.query(DagRun)
-                .filter(
-                    DagRun.dag_id == task_list[0].dag_id,
-                    DagRun.execution_date == execution_date,
+            dag_run = session.scalar(
+                select(DagRun).where(
+                    DagRun.dag_id == task_list[0].dag_id, DagRun.execution_date == execution_date
                 )
-                .one()
             )
+
```

Comment:
the .one() method in the v1 fails if there is no result, which is data validation check. For that we cannot replace it with `session.scalar` which returns None if there is no result.

According to [the migration doc](https://docs.sqlalchemy.org/en/20/changelog/migration_20.html):
the following query in v1:
```python
session.query(User).\
  filter_by(name="some user").\
  one()
```
could be written as the following in v2 style:
```python
session.execute(
  select(User).
  filter_by(name="some user")
).scalar_one()
```

Reply:
Or

```python
session.scalars(...).one()
```

Reply:
fixed it